pyproject.toml

```diff
@@ -28,7 +28,8 @@ dependencies = [
     "pandas",
     "scikit-image",
     "xarray",
-    "jinja2"
+    "jinja2",
+    "s3fs"
```

Comment:
I'd be slightly tempted to put "s3fs" in a separate, optional set of dependencies since it's not required for the core usage of the package, e.g. under `[project.optional.dependencies]` below in a set called "s3" or similar.
Where s3 is imported in the modules this would require either: the import to be only inside the sections that use it so it isn't imported unless required; or at the top of the module in a `try`/`except` block; or we make a separate s3 module which is the only place it's imported.

Reply:
Agreed, I've moved it to optional dependencies.

"s3fs" is only imported in blocks where it is used. Creates a (very small) amount of repetition but because the functionality sits within the `ASLICalculator` class, I think this is the simplest way to achieve this.